src/tests/transaction-flow.ts

```diff
@@ -0,0 +1,291 @@
+const zkAppKey = (await Lightnet.acquireKeyPair()).privateKey;
+const zkAppAddress = zkAppKey.toPublicKey();
+
+// Same balance as remote ledger
+const balance = (1550n * 10n ** 9n).toString();
+Local.addAccount(sender, balance);
+Local.addAccount(zkAppAddress, balance);
+
+console.log('Compiling the smart contract.');
+const { verificationKey } = await SimpleZkapp.compile();
+const zkApp = new SimpleZkapp(zkAppAddress);
+console.log('');
+
+console.log('Testing network auxiliary functions do not throw');
+await testLocalAndRemote(async () => {
+  try {
+    await Mina.transaction({ sender, fee: transactionFee }, () => {
+      Mina.getNetworkConstants();
+      Mina.getNetworkState();
+      Mina.getNetworkId();
+      Mina.getProofsEnabled();
+    });
+  } catch (error) {
+    assert.ifError(error);
+  }
+});
+console.log('');
+
+console.log(
+  `Test 'fetchAccount', 'getAccount', and 'hasAccount' match behavior using publicKey: ${zkAppAddress.toBase58()}`
+);
+await testLocalAndRemote(async () => {
+  try {
+    await fetchAccount({ publicKey: zkAppAddress }); // Must call fetchAccount to populate internal account cache
+    const account = Mina.getAccount(zkAppAddress);
+    return {
+      publicKey: account.publicKey,
+      nonce: account.nonce,
+      hasAccount: Mina.hasAccount(zkAppAddress),
+    };
+  } catch (error) {
+    assert.ifError(error);
+  }
+});
+console.log('');
+
+console.log('Test deploying zkApp for public key ' + zkAppAddress.toBase58());
+await testLocalAndRemote(async () => {
+  try {
+    const transaction = await Mina.transaction(
+      { sender, fee: transactionFee },
+      () => {
+        zkApp.deploy({ verificationKey });
+      }
+    );
+    transaction.sign([senderKey, zkAppKey]);
+    await sendAndVerifyTransaction(transaction);
+  } catch (error) {
+    assert.ifError(error);
+  }
+});
+console.log('');
+
+console.log("Test calling 'update' method on zkApp does not throw");
+await testLocalAndRemote(async () => {
+  try {
+    const transaction = await Mina.transaction(
+      { sender, fee: transactionFee },
+      () => {
+        zkApp.update(Field(1), PrivateKey.random().toPublicKey());
+      }
+    );
+    transaction.sign([senderKey, zkAppKey]);
+    await sendAndVerifyTransaction(transaction);
+    await Mina.fetchEvents(zkAppAddress, TokenId.default);
+  } catch (error) {
+    assert.ifError(error);
+  }
+});
+console.log('');
+
+console.log("Test specifying 'invalid_fee_access' throws");
+await testLocalAndRemote(async () => {
+  let errorWasThrown = false;
+  try {
+    const transaction = await Mina.transaction(
+      { sender, fee: transactionFee },
+      () => {
+        AccountUpdate.fundNewAccount(zkAppAddress);
+        zkApp.update(Field(1), PrivateKey.random().toPublicKey());
+      }
+    );
+    transaction.sign([senderKey, zkAppKey]);
+    await sendAndVerifyTransaction(transaction);
+  } catch (error) {
+    errorWasThrown = true;
+  }
+  assert(errorWasThrown);
```

Comment:
maybe use `assert.throws(() => {...})` to make this read cleaner?

Reply:
`assert.throws()` doesn't seem to play well with async functions :sob: I think `assert.rejects` fulfills the same role though!

Reply:
oh sorry yeah!